javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -711,7 +708,12 @@ public Node(T value, List<Node<T>> children) {
             Objects.requireNonNull(children, "children is null");
             this.value = value;
             this.children = children;
-            this.size = 1 + this.children.foldLeft(0, (acc, child) -> acc + child.size);
+
```

Comment:
expanding the `foldLeft` and getting rid of the lambda in this critical section made a measurable difference